File: backend/app/models/isp_inventory.py
from datetime import datetime
from typing import Optional
from app.schemas.enums import EquipmentCategory, EquipmentStatus
import strawberry

@strawberry.type
class DBISPInventory:
    _id: str
    name: str
    category: EquipmentCategory
    organizationId: str
    model: Optional[str]
    manufacturer: Optional[str]
    serialNumber: Optional[str]
    macAddress: Optional[str]
    ipAddress: Optional[str]
    quantity: int
    quantityThreshold: Optional[int]
    unitPrice: float
    status: EquipmentStatus
    location: Optional[str]
    assignedTo: Optional[str]
    warrantyExpirationDate: Optional[datetime]
    purchaseDate: Optional[datetime]
    specifications: Optional[str]
    notes: Optional[str]
    createdAt: datetime
    updatedAt: datetime

File: frontend/types/isp_inventory.ts
export type EquipmentCategory = 
  | 'ROUTER'
  | 'SWITCH'
  | 'ACCESS_POINT'
  | 'ANTENNA'
  | 'CABLE'
  | 'CONNECTOR'
  | 'POWER_SUPPLY'
  | 'SERVER'
  | 'CPE'
  | 'TOOLS'
  | 'OTHER';

export const EquipmentStatus = {
  AVAILABLE: 'AVAILABLE',
  IN_USE: 'IN_USE',
  DEFECTIVE: 'DEFECTIVE',
  IN_REPAIR: 'IN_REPAIR',
  RESERVED: 'RESERVED',
  DISPOSED: 'DISPOSED'
} as const;

export type EquipmentStatus = typeof EquipmentStatus[keyof typeof EquipmentStatus];

export type ISPInventory = {
  id: string;
  name: string;
  category: EquipmentCategory;
  organization: {
    id: string;
    name: string;
  };
  model?: string;
  manufacturer?: string;
  serialNumber?: string;
  macAddress?: string;
  ipAddress?: string;
  quantity: number;
  quantityThreshold?: number;
  unitPrice: number;
  status: EquipmentStatus;
  location?: string;
  assignedTo?: string;
  warrantyExpirationDate?: string;
  purchaseDate?: string;
  specifications?: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
};

export interface ISPInventoriesResponse {
  inventories: {
    success: boolean;
    message: string;
    inventories: ISPInventory[];
  };
}

export interface ISPInventoryResponse {
  inventory: {
    success: boolean;
    message: string;
    inventory: ISPInventory;
  };
}

export type CreateISPInventoryInput = {
  name: string;
  category: EquipmentCategory;
  organizationId: string;
  model?: string;
  manufacturer?: string;
  serialNumber?: string;
  macAddress?: string;
  ipAddress?: string;
  quantity: number;
  quantityThreshold?: number;
  unitPrice: number;
  location?: string;
  warrantyExpirationDate?: string;
  purchaseDate?: string;
  specifications?: string;
  notes?: string;
};

export type UpdateISPInventoryInput = Partial<CreateISPInventoryInput> & {
  status?: EquipmentStatus;
  assignedTo?: string;
};

